src/pcre2_compile.c

```diff
@@ -1863,7 +1863,12 @@ else
         but we keep it just to be safe and because it will also catch the bogus
         value set on failure of that function. */
 
-        if ((unsigned)s > MAX_GROUP_NUMBER) *errorcodeptr = ERR61;
+        if ((unsigned)s > MAX_GROUP_NUMBER)
+          {
+          if (s != INT_MAX)
```

Comment:
this could be written instead as `PCRE2_ASSERT(s == INT_MAX)`

Reply:
Yep, Zoltan already pointed that out and I pushed updated code accordingly.

Reply:
Hmm, looks like an earlier version of my commit was squashed into this PR. Definitely it is better to use `PCRE2_ASSERT(s == INT_MAX)`; could you update that?

Reply:
That is what happens when you `commit --amend` and `git push --force` without a PR, while providing commit references through comments.